core/field_variable.js

```diff
@@ -224,6 +224,43 @@ Blockly.FieldVariable.prototype.getVariableTypes_ = function() {
   return variableTypes;
 };
 
+/**
+ * Parse the optional arguments representing the allowed variable types and the
+ * default variable type.
+ * @param {Array.<string>=} opt_variableTypes A list of the types of variables
+ *     to include in the dropdown.  If null or undefined, variables of all types
+ *     will be displayed in the dropdown.
+ * @param {string=} opt_defaultType The type of the variable to create if this
+ *     field's value is not explicitly set.  Defaults to ''.
+ * @private
+ */
+Blockly.FieldVariable.prototype.setTypes_ = function(opt_variableTypes,
+    opt_defaultType) {
+  // If you expected that the default type would be the same as the only entry
+  // in the variable types array, tell the Blockly team by commenting on #1499.
+  this.defaultType_ = opt_defaultType || '';
+  // Set the allowable variable types.  Null means all types on the workspace.
+  if (opt_variableTypes == null || opt_variableTypes == undefined) {
+    this.variableTypes = null;
+  } else if (Array.isArray(opt_variableTypes)) {
+    this.variableTypes = opt_variableTypes;
+    // Make sure the default type is valid.
+    var isInArray = false;
+    for (var i = 0; i < this.variableTypes.length; i++) {
+      if (this.variableTypes[i] == this.defaultType_) {
+        isInArray = true;
+      }
+    }
+    if (!isInArray) {
```

Comment:
Do you think it matters whether you throw the error before setting this.variableTypes?  I guess I'd expect the error to get thrown without having the side effect.

Reply:
Updated.